apps/src/JSInterpreter.js

```diff
@@ -813,6 +847,67 @@ JSInterpreter.prototype.createGlobalProperty = function (name, value, parent) {
 };
 
 /**
+ * Slightly modified version of interpreter's getValueFromScope. Does not
+ * throw an exception that can be caught by the interpreted program.
+ */
+JSInterpreter.prototype.getValueFromScope = function (name) {
+  var scope = this.interpreter.getScope();
+  while (scope) {
+    if (this.interpreter.hasProperty(scope, name)) {
+      return this.interpreter.getProperty(scope, name);
+    }
+    scope = scope.parentScope;
+  }
+  throw new ReferenceError('Unknown identifier: ' + name);
+};
+
+/**
+ * Returns an interpreter object representing the member expression.
+ */
+JSInterpreter.prototype.getValueFromMemberExpression = function (expression) {
+  var object;
+  if (expression.object.type === 'MemberExpression') {
+    object = this.getValueFromMemberExpression(expression.object);
+  } else if (expression.object.type === 'Identifier') {
+    object = this.getValueFromScope(expression.object.name);
+  } else {
+    throw new Error("Unsupported expression object");
+  }
+  if (expression.property.type === 'Identifier') {
+    return this.interpreter.getValue([object, expression.property.name]);
+  } else if (expression.property.type === 'Literal') {
+    return this.interpreter.getValue(
+        [object, this.interpreter.createPrimitive(expression.property.value)]);
+  }
+};
+
+/**
+ * Evaluate watch variable based on current scope.
+ */
+JSInterpreter.prototype.evaluateWatchVariable = function (watchVar) {
+  var value;
+  try {
+    var ast = window.acorn.parse(watchVar);
+    if (ast.type === 'Program' &&
+        ast.body[0].type === 'ExpressionStatement') {
+      var expressionNode = ast.body[0].expression;
+      if (expressionNode.type === 'Identifier') {
+        value = this.getValueFromScope(expressionNode.name);
+      } else if (expressionNode.type === 'MemberExpression') {
+        value = this.getValueFromMemberExpression(expressionNode);
```

Comment:
This if-else exactly mirrors one in `getValueFromMemberExpression` above.  Could extract a helper, or somehow unify the two so you always call `getValueFromMemberExpression` here and let it care about type.